### ISSUE TITLE
Tests for core business logics

### ISSUE THREAD

**Issue by andrzejWilde** [lootek/go-port-service#1]
The codebase contains tests for simple model-to-model transformations yet it lacks tests for core business logics. 
Could you add tests that would verify correctness of the main functionality, please?

Please have a think should this be accomplished by extra unit tests, integration tests or perhaps both.

**Comment by lootek** [comment]
Yeah, you're right there should've been more tests (as always ;) ). And those were beneficial even at the time of writing as I noticed I was initially not very strict with the requirements - so I've changed the code to expose Upsert() (instead of Add()/Update()). Plus it was more obvious and kind of natural when called from the HTTP handler. The next thing to do is to add test cases for the JSON file upload there which I don't have time for right now.

The next major thing to do that's worth adding is a high-level integration test to be run in the dockerized runtime environment (most probably using sth like a python helper script). Its logic would be very simple, yet needs some additional work to set things up:
* upload some trash to test the app resiliency
* upload a simple (short) file to test the basic functionality
* upload the same file twice to test if the content gets updated (instead of added indefinitely)
* upload 2 files with slightly changed content
* upload a _big_ file (and test if the container does not exceed some memory limit)

I also added some TODO notes for the future in the code.

I wish I had more time to spend on this and polish it to be a really nice and clean e2e-ready codebase.

**Comment by lootek** [comment]
@andrzejWilde, yeah, you're right there should've been more tests (as always ;) ). And those were beneficial even at the time of writing as I noticed I was initially not very strict with the requirements - so I've changed the code to expose Upsert() (instead of Add()/Update()). Plus it was more obvious and kind of natural when called from the HTTP handler. The next thing to do is to add test cases for the JSON file upload there which I don't have time for right now.

The next major thing to do that's worth adding is a high-level integration test to be run in the dockerized runtime environment (most probably using sth like a python helper script). Its logic would be very simple, yet needs some additional work to set things up:
* upload some trash to test the app resiliency
* upload a simple (short) file to test the basic functionality
* upload the same file twice to test if the content gets updated (instead of added indefinitely)
* upload 2 files with slightly changed content
* upload a _big_ file (and test if the container does not exceed some memory limit)

I also added some TODO notes for the future in the code.

I wish I had more time to spend on this and polish it to be a really nice and clean e2e-ready codebase.
